app/src/modelProviders/fine-tuned/getCompletion-2.ts

```diff
@@ -34,27 +36,42 @@ export async function getCompletion2(
   const prunedInput = { messages: prunedMessages, ...omit(input, "messages") };
 
   if (fineTune.provider === "openai") {
-    const model = fineTune.openaiModelId;
+    return getOpenAIFineTuneCompletion(fineTune, prunedInput);
+  } else {
+    return getModalCompletion(fineTune, prunedInput);
+  }
+}
 
-    if (!model) throw new Error("No OpenAI model ID found");
+async function getOpenAIFineTuneCompletion(
+  fineTune: FineTune,
+  input: ChatCompletionCreateParams,
+): Promise<ChatCompletion | Stream<ChatCompletionChunk>> {
+  const model = fineTune.openaiModelId;
 
-    // TODO: create pipeline without this conversion once OpenAI supports tool_calls for their fine-tuned models
+  if (!model) throw new Error("No OpenAI model ID found");
 
+  if (input.stream) {
     const completion = await getOpenaiCompletion(fineTune.projectId, {
-      // convert tool call input to function call input
-      ...convertToolCallInputToFunctionInput(prunedInput),
+      ...input,
       model,
     });
-    if (completion.choices[0]?.message && input.tools?.length) {
-      // convert function call output to tool call output
-      completion.choices[0].message = convertFunctionMessageToToolCall(
-        completion.choices[0].message,
-      ) as ChatCompletionMessage;
-    }
     return completion;
-  } else {
-    return getModalCompletion(fineTune, prunedInput);
   }
+
+  // TODO: create pipeline without this conversion once OpenAI supports tool_calls for their fine-tuned models
+  input = convertToolCallInputToFunctionInput(input);
```

Comment:
I'm a little confused about what's going on here. Why are we converting tool calls only for non-streamed responses? Shouldn't tool calls be treated the same way independent of whether you're streaming or not?

Reply:
I added a comment earlier in the file to clarify. Basically converting a streamed function call into a tool call output on the fly would be complicated.